package.json

```diff
@@ -1,79 +1,102 @@
 {
   "name": "prismarine-web-client",
-  "version": "1.5.0",
+  "version": "0.0.0-dev",
   "description": "A minecraft client running in a browser",
-  "main": "index.js",
   "scripts": {
-    "postinstall": "node scripts/patchPackages.js",
-    "build": "webpack --config webpack.prod.js",
-    "build-dev": "webpack --config webpack.dev.js",
-    "start": "node --max-old-space-size=8192 server.js 8080 dev",
+    "start": "node scripts/build.js copyFilesDev && node scripts/prepareData.mjs && node esbuild.mjs --watch",
+    "start-watch-script": "nodemon -w esbuild.mjs esbuild.mjs",
+    "build": "node scripts/build.js copyFiles && node scripts/prepareData.mjs -f && node esbuild.mjs --minify --prod",
+    "check-build": "tsc && pnpm build",
+    "test:cypress": "cypress run",
+    "test:e2e": "start-test http-get://localhost:8080 test:cypress",
     "prod-start": "node server.js",
-    "build-dev-start": "npm run build-dev && npm run prod-start",
-    "build-start": "npm run build && npm run prod-start",
-    "prepublishOnly": "npm run build",
-    "lint": "standard",
-    "fix": "standard --fix",
-    "test": "npm run lint && mocha"
-  },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/PrismarineJS/prismarine-web-client.git"
+    "postinstall": "node scripts/gen-texturepack-files.mjs",
+    "lint": "eslint \"{src,cypress}/**/*.{ts,js,jsx,tsx}\""
   },
   "keywords": [
     "prismarine",
     "web",
     "client"
   ],
+  "bin": "./server.js",
   "author": "PrismarineJS",
   "license": "MIT",
-  "bin": {
-    "prismarine-web-client": "./server.js"
-  },
-  "bugs": {
-    "url": "https://github.com/PrismarineJS/prismarine-web-client/issues"
-  },
-  "homepage": "https://github.com/PrismarineJS/prismarine-web-client#readme",
   "dependencies": {
+    "@dimaka/interface": "0.0.3-alpha.0",
+    "@types/react": "^18.2.20",
```

Comment:
Why react ?

Reply:
I'm not the type of guy who just decided to make a full rewrite using my favorite frameworks. I removed lit in the latest version not only because React has a much greater ecosystem (such as good storybook support) & clean function components and I didn't see a single reason to use lit. I could adopt, really, but there was a thing that was blocking a lot of stuff. The main reason why I had to drop lit is the *shadow dom* (e.g. no way of using global CSS variables).
I've enjoyed the rewrite and am happy with the final result, but if you're not I'm sorry! I actually didn't see other solutions.

Reply:
> no way of using global CSS variables

this is one of the best feature of lit

Reply:
> this is one of the best feature of lit

How?

This is the same as saying that "no way to build scalable applications is also the best feature of lit". These kinds of restrictions don't make code more maintainable it just makes it harder to find solutions. Meanwhile react has the best ecosystem you can find if you're a frontend developer, it still is the fastest & easiest way to build robust UI at scale. Just take the current storybook integration: zero-config first-class react support. I tried to adopt lit, but it sucks at absolutely everything, no good state management, no hmr, plugins and so on... Also, my team could easily work with react + ts and even build new components without any problems so I don't see objective points here, for most other devs it doesn't really make the difference what framework you use...

Reply:
global css makes it impossible to build reusable components, it's not much more complicated than that

that's why react doesn't really have reusable components but instead it has sets of components you need to all use together

beyond that, regarding what framework to use here, I am not yet convinced we need anything big as react / state management / etc

Reply:
I guess the only way forward here is for me to go and pick stuff that makes sense from your fork, and eventually we converge to something reasonable

Reply:
But React is not that big and at some points even easier to learn. You just need to understand how Function components work and that's. And what is more reasonable from your point? This is an absolutely fantastic & simple UI framework (any frontend guy would say you the same), I don't understand what's the issue with that (sorry)